includes/subscriptions/class-wc-payments-subscription-service.php

```diff
@@ -736,8 +736,10 @@ public function get_recurring_item_data_for_subscription( WC_Subscription $subsc
 		foreach ( $subscription->get_items() as $item ) {
 			$product = $item->get_product();
 
-			if ( ! WC_Subscriptions_Product::is_subscription( $product ) ) {
-				continue;
+			// We always need a WCPay product ID for all subscription line items.
+			// If the product isn't a subscription product, calling $product_service->get_wcpay_product_id() won't create one on the fly so we need to create one manually.
+			if ( ! WC_Subscriptions_Product::is_subscription( $product ) && ! $this->product_service::has_wcpay_product_id( $product ) ) {
```

Comment:
Since we're adding all the items in the subscription despite it's a subscription product or not, shouldn't `has_wcpay_product_id()` check alone suffice?

```suggestion
			if ( ! $this->product_service::has_wcpay_product_id( $product ) ) {
```

Reply:
Yeah it would. 

I was thinking along the lines of targeting the specific circumstance where the product isn't a subscription and so can't get over [this condition](https://github.com/Automattic/woocommerce-payments/blob/fix/3826/includes/subscriptions/class-wc-payments-product-service.php#L112) and to create one when we attempt to fetch it. ie only non-subscription products without a wcpay product ID need to handled separately. 

But I'm happy to just go with `$this->product_service::has_wcpay_product_id()`.

Reply:
Hey @shendy-a8c what do you think of [59d0f27](https://github.com/Automattic/woocommerce-payments/pull/3860/commits/59d0f27916f8a6b4f19e7c6e4ecddf6abb29cbf6) instead?

Reply:
Sorry for the delay, @james-allan. Looks good!